src/components/Cart.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState, useEffect, useContext, useRef } from 'react'
+import { CartInfoContext } from "../providers/CartInfoProvider";
+
+const Cart = () => {
+    const { _confirmedOrdersInfo, _updateCartInfo } = useContext(CartInfoContext);
+
+    const [confirmedOrdersInfo, setConfirmedOrdersInfo] = _confirmedOrdersInfo;
+    const [updateCartInfo, setUpdateCartInfo] = _updateCartInfo
+
+
+    // checking and getting the items from local storage and storing them into the state context value of 'confirmedOrdersInfo' onmount
+    // this will check if there is anything in the local storage. If there are values stored in the local storage, then stored them into confirmedOrdersInfo
+    useEffect(() => {
+        if (JSON.parse(localStorage.getItem("confirmed orders") !== null)) {
+            setConfirmedOrdersInfo(JSON.parse(localStorage.getItem("confirmed orders")));
+            setUpdateCartInfo(true);
+        };
+    }, []);
+
+    // in order to avoid unnecessary re-renderings by using a state value to get the values below, the useRef will preserve the values through each re-rendering and will update these values when the 'updateCartInfo' is set to true
+    const cartQuantityTotal = useRef(0);
+    const cartPriceTotal = useRef(0);
+
```

Comment:
You need to display the total value and quantity once that change, so obviously you need a rerender.
I don't understand how `useRef` helps here?
Also, you keep doing what I told you not to do: use 2 different react variables when one is a derivative of the other.

Reply:
So, I can't have any react variables to store the price and quantity total of the cart?  My reasoning for using variables was to display the total price and quantity onto the DOM. I can just put `confirmedOrdersInfo.map((order) => parseFloat(order.totalOrderPrice)).reduce((numA, numB) => numA + numB);` directly into the JSX code in order to display those values. But that seems to be a bit clunky

Reply:
Your cart should be an array, so to to show the number of times you should only use `cart.length`

Reply:
But each value--the order--in the array--which will be an object--will have its own quantity.  Each order will be structured as follows: `{
        id: null,
        restaurantName: null,
        infoOfMainMeatItem: null,
        confirmedAddOnsToOrder: null,
        allAddOns: null,
        confirmedOrderQuantity: 0,
        totalMeatPrice: 0,
        totalConfirmedAddOnPrice: 0,
        totalOrderPrice: 0
    }`. So that's why I need to map through the array to get the values under `confirmedOrderQuantity` and find the total sum. That sum will be the total quantity of the cart and will be displayed on the cart icon.